File: chapter6/python/6-5/client65.py
from socket import *
from tcpPacket import *
import sys
import pickle

ip = sys.argv[1]
host = sys.argv[2]
address = (ip, int(host))
print("target IP: {}, port: {}".format(ip, host))

tcpClientSocket  = socket(AF_INET, SOCK_STREAM)
tcpClientSocket.connect(address)
start = 1

while True:
    if start == 1:
        pkt = getpkg(0, 0, 1, 0)
        print("request for connection: syn({}), ack({})".format(pkt.syn, pkt.ack))
        tcpClientSocket.send(pickle.dumps(pkt))
        start = 0
    data = tcpClientSocket.recv(1024)
    if not data:
        break    
    pkt = pickle.loads(data)
    print("reply for connection: syn({}), ack({})".format(pkt.syn, pkt.ack))
    
tcpClientSocket.close()

File: chapter6/c++/6-5-c/client65.cpp
#include <iostream>
#include "TCPPacket.h"

using namespace std;

int main(int argc, char **argv) {
    int c;
    string ip;
    int port;
    ip = string(argv[1]);
    port = atoi(argv[2]);
    cout << "target IP: " << ip << " , port: " << port << endl;
    TCPPacket packet(9999);
    packet.conn(ip, port);

    return 0;
}